xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/renderer/xhtml/link/AbstractXHTMLLinkTypeRenderer.java

```diff
@@ -221,7 +226,10 @@ public void endLink(ResourceReference reference, boolean freestanding, Map<Strin
     {
         // If there was no link content then generate it based on the passed reference
         if (!hasLabel()) {
-            getXHTMLWikiPrinter().printXMLStartElement(SPAN, new String[][] { { CLASS, "wikigeneratedlinkcontent" } });
+            getXHTMLWikiPrinter().printXMLStartElement(SPAN, new String[][]
+            {
+                { CLASS, "wikigeneratedlinkcontent" }
```

Comment:
Is it possible to already has classes defined at this point?

Reply:
_Note:_ the changes here are purely quality of code, I found it hard to read out so I separated it on a few lines. This part was particularly interesting to me because I wanted to replicate a behavior similar to the call to computeLabel on line 233. 

The class defined here is the one we want for the label of the link, which is a new child element of the link. We print this label completely on these few lines.